src/values/ray-query.cxx

```diff
@@ -0,0 +1,290 @@
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionType(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate ray to intersection distance.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return distance to the intersection.
+    float getIntersectionT(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionT(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection instance custom index.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return instance custom index.
+    int getIntersectionInstanceCustomIndex(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionInstanceCustomIndex(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection instance id.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return instance id.
+    int getIntersectionInstanceId(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionInstanceId(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection instance shader binding table (SBT) record offset.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return instance SBT record offset.
+    unsigned getIntersectionInstanceShaderBindingTableRecordOffset(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionInstanceShaderBindingTableRecordOffset(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection geometry index.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return geometry index.
+    int getIntersectionGeometryIndex(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionGeometryIndex(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection primitive index.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return primitive index.
+    int getIntersectionPrimitiveIndex(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionPrimitiveIndex(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection barycentrics.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return barycentric coordinates.
+    glm::vec2 getIntersectionBarycentricsGLM(const unsigned intersection) const {
```

Comment:
Personally, I would fold this and the other get*GLM methods into their get* counterparts. Or at least, do these need to be public? Also, they are currently only called by their get* counterpart, so I don't know if they justify their independence.

Reply:
Decided to merge getGLM into get methods.